File: bot/events/ready.js
'use strict'

import 'dotenv/config'
import { Client, Events, ChannelType, GuildMemberManager } from 'discord.js'
import { calendar } from '../helpers/import-calendar.js'
import { checkExisitingForumEvents } from '../helpers/thread-manager.js'
import startCalendarMonitor from '../daemons/calendar-monitor.js'

export default {
  name: Events.ClientReady,
  once: true,
  execute: async (client) => {
    try {
      
      // get our guild (the server)
      const guild = await client.guilds.cache.get(process.env.WUMPUS_GUILD)
      console.log(`[discord] logged into "${guild.name}" [${guild.id}] as "${client.user.username}#${client.user.discriminator}"`)
      
      console.log(`[discord] checking calendar data`)
      const courseCalendar = await calendar()
      
      // check ...
      await checkExisitingForumEvents(
        process.env.NETW,
        courseCalendar.courses,
        process.env.PROG_WUMPUS_FORUM_CHANNEL,
        ChannelType,
        client
      )

      // Start the calendar monitoring daemon
      await startCalendarMonitor(client)

      return true
    } catch (ex) {
      console.error(`[discord] bot unable to start`, ex)
    }
  },
}

File: bot/events/ready.py
"""Ready event handler for the bot."""

import os
import discord
from discord.ext import commands


class ReadyEvent(commands.Cog):
    """Ready event cog."""

    def __init__(self, bot):
        self.bot = bot

    @commands.Cog.listener()
    async def on_ready(self):
        """Handle bot ready event."""
        try:
            guild = self.bot.get_guild(int(os.getenv('WUMPUS_GUILD')))
            
            if guild:
                print(
                    f'[discord] logged into "{guild.name}" [{guild.id}] '
                    f'as "{self.bot.user.name}#{self.bot.user.discriminator}"'
                )
            else:
                print(f'[discord] logged in as "{self.bot.user.name}"')
            
            # Sync slash commands
            print('[discord] Syncing application commands...')
            await self.bot.tree.sync()
            print('[discord] Application commands synced')
            
            # TODO: Import calendar and check forum events
            # print('[discord] checking calendar data')
            # calendar_data = await calendar()
            # await check_existing_forum_events(...)
            
            # TODO: Start calendar monitoring daemon
            # await start_calendar_monitor(self.bot)
            
        except Exception as ex:
            print(f'[discord] bot unable to start: {ex}')


async def setup(bot):
    """Setup function to add the cog to the bot."""
    await bot.add_cog(ReadyEvent(bot))
